### PR TITLE
SV-70 Google Auth Setup

### PR DESCRIPTION
**Pull request by mackella**
# What?

 * Added integration test structure
 * Integrated node-lambda-authorizer
 * Added integration test cases
 * Refactored how vars are injected into serverless environment

# Why?

 * Enable access to the MaT v2 API by authorised users
 * SSO provides a better UX and improved security
 * Enables group-based access to both the overall app and potentially specific functionality

# Checklist

- [ ] Unit tests
- [x] Integration tests
- [ ] Cypress tests

# Notes

Please carefully review the changes to the serverless configuration and the integration testing framework.

# Next steps

  
Add CircleCI task to enable integration testing in staging.

### CODE REVIEW

File: jest.unit.config.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
+// For a detailed explanation regarding each configuration property, visit:
```
Comment (ccp92) [inline jest.unit.config.js:1]: Could probably clean the comments out of this file if they're just pre-generated

File: swagger.yml [hunk 2]
```diff
@@ -0,0 +1,222 @@
+swagger: '2.0'
```
Comment (ccp92) [inline swagger.yml:1]: Is this swagger autogenerated? I don't think it's picking up the 403 unauthorised responses (and do we want it do so)?

File: jest.unit.config.js (hunk 1 above)
Comment (mackella) [reply jest.unit.config.js:1]: Yes, agreed. Both jest config files should ultimately be reduced to only specifying overrides. I find it helpful to have the boilerplate there for reference while we're still tweaking the config though.

File: swagger.yml (hunk 2 above)
Comment (mackella) [reply swagger.yml:1]: No, this is the manually generated swagger file from the service design ticket. It needs to be updated as we implement each API resource. It'll be moved to Hackney's SwaggerHub once we get access.

Comment (mackella) [reply swagger.yml:1]: Not even sure why this file is in this PR. It was already merged via PR #7

**Pull request merged**